coding-principles.md

```diff
@@ -31,6 +31,63 @@ var myUniqueArray = _.uniq(myArray);
 
 One might argue that extensive use of this syntactic sugar will lead to a slower codebase, but modern JavaScript engines do a great job of optimising the code we write to the point that the readibility benefits outweigh any minor performance quibbles. In browsers especially, a huge amount more CPU time is lost to rendering than actual application logic.
 
+## Keep arguments to functions simple
+Only pass a few to each function. If you find there are too many, re-consider the function's purpose or pass an object. Too many arguments is often a [code smell](https://en.wikipedia.org/wiki/Code_smell) indicating that the function should be refactored into smaller functions. The goal is to keep it readable & re-usable:
+
+Bad:
+```javascript
+log( 1234, 201.99, 1, 4, 'Hotel', '/hotels', true );
+```
+
+Good:
+```javascript
+log({
+  sku: 1234,
+  price: 201.99,
+  quantity: 1,
+  position: 4,
+  type: 'Hotel',
+  page: '/hotels',
+  isPromoted: true
+});
+```
+
+In the "good" example we should be dealing with validating the object passed in or merging default properties thus making the code "self-documenting":
+
+```javascript
+function log(options) {
+  var requirements = ['sku', 'price', 'quantity'];
+  
+  // Check `requirements` against `options` passed in & handle accordingly.
+
+  var defaults = {
+    isPromoted: false
+  };
+  
+  // Merge `options` with `defaults` here
+  ... etc
+}
```

Comment:
@msaspence I wasn't too sure about confusing the point with full examples as if this were a "real" piece of code, you could use `lodash` to check props etc.

To be honest I struggled with what is a "good" function body for this, really the point is about discouraging function declarations that have so many arguments that it becomes hard to interpret the functions purpose.